weblab_cg/data/tests/chaste_reference_models/Normal/hodgkin_huxley_squid_axon_model_1952_modified.cpp

```diff
@@ -0,0 +1,170 @@
+		// Units: dimensionless; Initial value: 0.05
+		double var_chaste_interface__sodium_channel_h_gate__h = rY[2];
+		// Units: dimensionless; Initial value: 0.6
+		double var_chaste_interface__potassium_channel_n_gate__n = rY[3];
+		// Units: dimensionless; Initial value: 0.325
+		
+        const double var_leakage_current__g_L = 0.3; // milliS_per_cm2
+        const double var_membrane__E_R = -75.0; // millivolt
+        const double var_leakage_current__E_L = 10.613 + var_membrane__E_R; // millivolt
+        const double var_leakage_current__i_L = (-var_leakage_current__E_L + var_chaste_interface__membrane__V) * var_leakage_current__g_L; // microA_per_cm2
+        const double var_potassium_channel__E_K = -12.0 + var_membrane__E_R; // millivolt
+        const double var_potassium_channel__g_K = 36.0; // milliS_per_cm2
+        const double var_potassium_channel__i_K = pow(var_chaste_interface__potassium_channel_n_gate__n, 4.0) * (-var_potassium_channel__E_K + var_chaste_interface__membrane__V) * var_potassium_channel__g_K; // microA_per_cm2
+        const double var_sodium_channel__E_Na = 115.0 + var_membrane__E_R; // millivolt
+        const double var_sodium_channel__g_Na = 120.0; // milliS_per_cm2
+        const double var_sodium_channel__i_Na = pow(var_chaste_interface__sodium_channel_m_gate__m, 3.0) * (-var_sodium_channel__E_Na + var_chaste_interface__membrane__V) * var_sodium_channel__g_Na * var_chaste_interface__sodium_channel_h_gate__h; // microA_per_cm2
+        const double var_chaste_interface__i_ionic = var_sodium_channel__i_Na + var_potassium_channel__i_K + var_leakage_current__i_L; // uA_per_cm2
+        
+		const double i_ionic = var_chaste_interface__i_ionic;
+        EXCEPT_IF_NOT(!std::isnan(i_ionic));
+        return i_ionic;
+    }
+    
+    void Dynamichodgkin_huxley_squid_axon_model_1952_modifiedFromCellML::EvaluateYDerivatives(double var_chaste_interface__environment__time, const std::vector<double>& rY, std::vector<double>& rDY)
+    {
+        // Inputs:
+        // Time units: millisecond
+		double var_chaste_interface__membrane__V = (mSetVoltageDerivativeToZero ? this->mFixedVoltage : rY[0]);
+		// Units: millivolt; Initial value: -75.0
+		double var_chaste_interface__sodium_channel_m_gate__m = rY[1];
+		// Units: dimensionless; Initial value: 0.05
+		double var_chaste_interface__sodium_channel_h_gate__h = rY[2];
+		// Units: dimensionless; Initial value: 0.6
+		double var_chaste_interface__potassium_channel_n_gate__n = rY[3];
+		// Units: dimensionless; Initial value: 0.325
+        
+        // Mathematics
+		double d_dt_chaste_interface__membrane__V;
+		const double var_potassium_channel_n_gate__alpha_n = -0.01 * (65.0 + var_chaste_interface__membrane__V) / (-1.0 + 0.0015034391929775724 * exp(-0.1 * var_chaste_interface__membrane__V)); // per_millisecond
+		const double var_potassium_channel_n_gate__beta_n = 0.31919868225786585 * exp(0.0125 * var_chaste_interface__membrane__V); // per_millisecond

```

Comment:
Is it numbers like these that were causing all the trouble with the testing? Would it be possible to just manually update these reference files to avoid having complex testing code?

Reply:
Yes and no (as a few occasions caused by simplify are also in the generated code) but I've found a way round it it's "only" for the develop test anyway.